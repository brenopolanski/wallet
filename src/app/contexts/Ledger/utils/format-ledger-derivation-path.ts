import { LedgerDerivationScheme } from "app/contexts/Ledger/contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
