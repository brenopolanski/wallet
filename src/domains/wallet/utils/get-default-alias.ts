import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let sameCoinWalletsCount = sameCoinWallets.length || 1;

	while (profile.wallets().findByAlias(makeAlias(sameCoinWalletsCount))) {
		sameCoinWalletsCount++;
	}

	return makeAlias(sameCoinWalletsCount);
};
