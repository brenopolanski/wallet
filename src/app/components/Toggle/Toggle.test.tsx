import React from "react";
import { fireEvent, render, screen } from "utils/testing-library";

import { Toggle } from "./Toggle";

describe("Toggle", () => {
	it("should render", () => {
		const { container, asFragment } = render(<Toggle />);

		expect(container).toBeInTheDocument();
		expect(screen.getByRole("checkbox")).not.toHaveAttribute("checked", "");
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render disabled", () => {
		const { container, asFragment } = render(<Toggle disabled />);

		expect(container).toBeInTheDocument();
		expect(screen.getByRole("checkbox")).toBeDisabled();
		expect(screen.getByRole("checkbox")).not.toHaveAttribute("checked", "");
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render small", () => {
		const { container, asFragment } = render(<Toggle small />);

		expect(container).toBeInTheDocument();
		expect(screen.getByRole("checkbox")).not.toHaveAttribute("checked", "");
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render always on", () => {
		const { container, asFragment } = render(<Toggle alwaysOn />);

		expect(container).toBeInTheDocument();
		expect(screen.getByRole("checkbox")).not.toHaveAttribute("checked", "");
		expect(asFragment()).toMatchSnapshot();
	});

	it("should toggle checked", () => {
		const { asFragment } = render(<Toggle />);
		const toggle = screen.getByRole("checkbox");

		fireEvent.change(toggle, { target: { checked: true } });

		expect(toggle.checked).toBe(true);
		expect(asFragment()).toMatchSnapshot();
	});
});
