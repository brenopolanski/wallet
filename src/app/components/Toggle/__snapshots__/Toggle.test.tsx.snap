// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle should render 1`] = `
<DocumentFragment>
  <label
    class="Toggle__Wrapper-sc-6s3wfu-0 efNWJo"
  >
    <input
      class="Toggle__Input-sc-6s3wfu-1 WwplI"
      type="checkbox"
    />
    <div
      class="Toggle__Handle-sc-6s3wfu-2 cobHFC"
    >
      <span
        class="Toggle__HandleInner-sc-6s3wfu-3 ddepvC"
      />
    </div>
  </label>
</DocumentFragment>
`;

exports[`Toggle should render always on 1`] = `
<DocumentFragment>
  <label
    class="Toggle__Wrapper-sc-6s3wfu-0 efNWJo"
  >
    <input
      class="Toggle__Input-sc-6s3wfu-1 WwplI"
      type="checkbox"
    />
    <div
      class="Toggle__Handle-sc-6s3wfu-2 cobHFC"
    >
      <span
        class="Toggle__HandleInner-sc-6s3wfu-3 cIyTyk"
      />
    </div>
  </label>
</DocumentFragment>
`;

exports[`Toggle should render disabled 1`] = `
<DocumentFragment>
  <label
    class="Toggle__Wrapper-sc-6s3wfu-0 kdYGGX"
    disabled=""
  >
    <input
      class="Toggle__Input-sc-6s3wfu-1 WwplI"
      disabled=""
      type="checkbox"
    />
    <div
      class="Toggle__Handle-sc-6s3wfu-2 cobHFC"
    >
      <span
        class="Toggle__HandleInner-sc-6s3wfu-3 kaYzOJ"
        disabled=""
      />
    </div>
  </label>
</DocumentFragment>
`;

exports[`Toggle should render small 1`] = `
<DocumentFragment>
  <label
    class="Toggle__Wrapper-sc-6s3wfu-0 iggSjt"
  >
    <input
      class="Toggle__Input-sc-6s3wfu-1 WwplI"
      type="checkbox"
    />
    <div
      class="Toggle__Handle-sc-6s3wfu-2 jRjdQR"
    >
      <span
        class="Toggle__HandleInner-sc-6s3wfu-3 enZLFu"
      />
    </div>
  </label>
</DocumentFragment>
`;

exports[`Toggle should toggle checked 1`] = `
<DocumentFragment>
  <label
    class="Toggle__Wrapper-sc-6s3wfu-0 efNWJo"
  >
    <input
      class="Toggle__Input-sc-6s3wfu-1 WwplI"
      type="checkbox"
    />
    <div
      class="Toggle__Handle-sc-6s3wfu-2 cobHFC"
    >
      <span
        class="Toggle__HandleInner-sc-6s3wfu-3 ddepvC"
      />
    </div>
  </label>
</DocumentFragment>
`;
